### ISSUE TITLE
Azure Function App : Runtime not generating a project.lock.json

### ISSUE THREAD

**Issue by Amysno** [Azure/azure-functions-host#3539]
**Summary (because incoming TLDR)**
When creating a .NET Azure Function (under a "Function app" resource), in the online editor, I upload a project.json to add some NuGet package (like said in [MS Documentation](https://docs.microsoft.com/fr-fr/azure/azure-functions/functions-reference-csharp#using-nuget-packages)). Normally the files should have been uploaded and the runtime should have generated the project.lock.json to have all the dependencies.

 
I wanted to create a Azure Function that would be triggered by a MS Flow. This azure function should add a user to a specified Yammer group etc..

#### Repro steps

1. I created a new "Function app" resource, specified .NET as runtime stack.

2. Once the resource was deployed, I created a new Function, and as Programming Environment, I chose "In-Portal".

3. And then, I chose the "Webhook + API". (strange, I could not choose the name for my function).

4. Without editing the function.json nor the run.csx, I try to upload my project.json which contains the following code : 
`{
    "frameworks": {
        "net46":{
            "dependencies": {
                "SharePointPnPCoreOnline": "2.22.1801"
            }
        }
    }
}`

5. I added the `using Microsoft.SharePoint.Client;` in my run.csx and tried to run the function.
and then, I got issues ...

#### Expected behavior
The runtime should generate project.lock.json with all my NuGet Package and dependencies references. And it should link it to my usings.

#### Actual behavior 
I got several errors such as : 

> 2018-10-01T10:01:06.457 [Error] Function compilation error
> 2018-10-01T10:01:06.496 [Error] run.csx(7,17): error CS0234: The type or namespace name 'SharePoint' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?)

And no project.lock.json had been generated.

**What did I try :** 
I tried several things : restart the resource, uploading a new project.json, uploading my project.lock.json (from my dev environement), creating everything from scratch again...
But nothing worked, and no project.lock.json was generated nor taken into account.
I thought maybe it did not work on a free trial account, but even with creating it with a real plan, nothing worked either.

#### Investigative information

Please provide the following:

- Timestamp: ? (how do you find it ?)
- Function App version (1.0 or 2.0): 2.0
- Function App name: Patatous (I know, it was a test one)
- Function name(s) (as appropriate): HttpTriggerCSharp1
- Invocation ID:
- Region: West Europe
- Runtime Version : 2.0.12115.0(~2)

**Comment by JocaPC** [comment]
I have the same issue. Did you resolved the problem?

**Comment by JohnReeson** [comment]
See this [SO thread](https://stackoverflow.com/a/51650086/9191057). [Doc enhancement](https://github.com/MicrosoftDocs/azure-docs/issues/13301#issuecomment-413926061) is in progress.

If your function runtime is ~2 on .NET Core, add references using `function.proj`

```
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFramework>netstandard2.0</TargetFramework>
    </PropertyGroup>
    <ItemGroup>
        <PackageReference Include="SharePointPnPCoreOnline" Version="3.2.1810"/>
    </ItemGroup>
</Project>

**Comment by Amysno** [comment]
Hi @JocaPC and @JohnReeson ,
Sorry for my answer's delay.
Finally found the issue !
While hosting the Ignite event, MS and Azure launched lots of new features.
One of them was the new runtime version of Azure function, which is not taking project.json into account anymore (as @JohnReeson said, now you'll have to use function.proj to declare your dependencies)

Because we had to run a azure function developped on previous runtime and deploy it to another tenant, we wanted to keep the previous runtime version and the previous architecture.
_@JohnReeson : thanks for your answer, we tried that but ended with lots of errors because of the dependencies, and did not want to redo everything, because we were trying to deploy a dev on a prod environment._

Here is how to do it : 

- Create your new Azure Function App

- then, go to the Function app settings  and find the Runtime version.

- Switch it to 1 (see the screenshot below  found on the Internet)
![image](https://user-images.githubusercontent.com/5112089/48064897-80c6fa00-e1c9-11e8-9982-7dd81287e5f6.png)

- Then, upload the project.json and make your function run and the magic happens !

Simple as that ! ;)
Sorry for the unclear and late answer, hope this will help you @JocaPC !
